src/crypto/SessionKeystore.h

```diff
@@ -0,0 +1,103 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace Crypto {
+
+/**
+ * @brief Interface for deriving and managing lifecycle of session keys.
+ *
+ * This interface provides an abstraction that allows to store session keys in a secure
+ * environment. It uses the concept of key handles that isolate the application from the actual
+ * key material.
+ */
+class SessionKeystore
+{
+public:
+    virtual ~SessionKeystore() {}
+
+    /**
+     * @brief Import raw key material and return key handle.
+     *
+     * @note This method hould only be used when using the raw key material in the Matter stack
+     * cannot be avoided. Ideally, crypto interfaces should allow platforms to perform all the
+     * cryptographic operations in a secure environment.
+     *
+     * If the function returns no error, the application is responsible for destroying the handle
+     * using DestroyKey() method when the key is no longer needed.
+     */
+    virtual CHIP_ERROR CreateKey(const Aes128KeyByteArray & keyMaterial, Aes128KeyHandle & key) = 0;
+
+    /**
+     * @brief Derive key from shared secret.
+     *
+     * Use HKDF as defined in the Matter specification to derive an AES key from the shared secret.
+     *
+     * TODO: Replace P256ECDHDerivedSecret with an opaque HKDF master key.
```

Comment:
Will this actually be possible?

Reply:
Technically yes, though may require changes to the session resumption storage mechanism so we'll need to weight pros and cons.